internal/core/metadata/rest_deviceprofile.go

```diff
@@ -62,8 +62,18 @@ func restAddDeviceProfile(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if Configuration.Writable.EnableValueDescriptorManagement {
+		// Check if the device profile name is unique so that we do not create ValueDescriptors for a DeviceProfile that
+		// will fail during the creation process later on.
+		nameOp := device_profile.NewGetProfileName(dp.Name, dbClient)
+		_, err := nameOp.Execute()
+		// The operator will return an ItemNotFound error if the DeviceProfile can not be found.
+		if err == nil {
+			http.Error(w, "Duplicate name for device profile", http.StatusConflict)
+			return
+		}
+
 		op := device_profile.NewAddExecutor(r.Context(), vdc, LoggingClient, dp.DeviceResources...)
```

Comment:
This should be `device_profile.ValueDescriptorAddExecutor` because as it is now, it reads as if it's a DeviceProfile Add executor.